tests/js/setup.ts

```diff
@@ -219,6 +211,89 @@ const routerFixtures = {
 };
 
 type TestStubTypes = TestStubFixtures & typeof routerFixtures;
+const jsFixturesDirectory = path.resolve(__dirname, '../../fixtures/js-stubs/');
+
+const extensions = ['.js', '.ts', '.tsx', '.json'];
+
+const specialMapping = {
+  AllAuthenticators: 'authenticators.js',
+  OrgRoleList: 'roleList.js',
+  MetricsField: 'metrics.js',
+  EventsStats: 'events.js',
+  DetailedEvents: 'events.js',
+  Events: 'events.js',
+  OutcomesWithReason: 'outcomes.js',
+  SentryAppComponentAsync: 'sentryAppComponent.js',
+  EventStacktraceMessage: 'eventStacktraceException.js',
+  MetricsTotalCountByReleaseIn24h: 'metrics.js',
+  MetricsSessionUserCountByStatusByRelease: 'metrics.js',
+  MOCK_RESP_VERBOSE: 'ruleConditions.js',
+  SessionStatusCountByProjectInPeriod: 'sessions.js',
+  SessionUserCountByStatusByRelease: 'sessions.js',
+  SessionUserCountByStatus: 'sessions.js',
+  SessionStatusCountByReleaseInPeriod: 'sessions.js',
+  SessionsField: 'sessions.js',
+  ProviderList: 'integrationListDirectory.js',
+  BitbucketIntegrationConfig: 'integrationListDirectory.js',
+  GithubIntegrationConfig: 'integrationListDirectory.js',
+  OrgOwnedApps: 'integrationListDirectory.js',
+  PublishedApps: 'integrationListDirectory.js',
+  SentryAppInstalls: 'integrationListDirectory.js',
+  PluginListConfig: 'integrationListDirectory.js',
+  DiscoverSavedQuery: 'discover.js',
+  VercelProvider: 'vercelIntegration.js',
+  TagValues: 'tagvalues.js',
+};
+
+function tryRequire(dir: string, name: string): any {
+  if (specialMapping[name]) {
+    return require(path.resolve(dir, specialMapping[name]));
+  }
+  for (const ext of extensions) {
+    try {
+      return require(path.resolve(dir, lowercasefirst(name) + ext));
+    } catch (err) {
+      // ignore
+    }
+  }
+  throw new Error('Failed to resolve file');
+}
+
+function lowercasefirst(string) {
+  return string.charAt(0).toLowerCase() + string.slice(1);
+}
+
+const lazyFixtures = new Proxy(
+  {},
+  {
+    get(target, prop: string) {
+      if (target[prop]) {
+        return target[prop];
+      }
+      if (routerFixtures[prop]) {
+        return routerFixtures[prop];
+      }
+
+      try {
+        const maybeModule = tryRequire(jsFixturesDirectory, prop);
+        for (const exportKey in maybeModule) {
+          target[exportKey] = maybeModule[exportKey];
+        }
+      } catch (e) {
+        // beep-boop
```

Comment:
what does `beep-boop` mean? 😄

Reply:
Oh sorry, miss-clicked resolve :D it's just a robot like sounds. Best explanation I found was
>"Beep boop" is a very common and a rather popular [catchphrase](https://knowyourmeme.com/memes/catchphrases) that depicts noises and expressions of various mechanical beings, mainly, [robots](https://knowyourmeme.com/memes/cultures/robots-and-artificial-intelligences) and droids, but also other computerized or electo-mechanical entities.